### ISSUE TITLE
Win64 mxe build error -  missing object in instantiation of trading_window::perform_trade()

### ISSUE THREAD

**Issue by ProfoundDarkness** [CleverRaven/Cataclysm-DDA#51978]
### Describe the bug

Building on Ubuntu 20.04 via MXE with the following line:
make -j$((`nproc`+0)) CROSS="${PLATFORM_64}" CCACHE=1 RELEASE=1 SDL=1 TILES=1 SOUND=1 LTO= LOCALIZE=1 LINTJSON= ASTYLE= RUNTESTS= PCH= LANGUAGES=en 

Build fails with the following errors:
src/npctrade.cpp: In instantiation of 'trading_window::perform_trade(npc&, const string&)::<lambda(item_pricing&, auto:30&&, bool)> [with auto:30 = trading_window::perform_trade(npc&, const string&)::<lambda(item_pricing&, auto:30&&, bool)>&]':
src/npctrade.cpp:763:53:   required from here
src/npctrade.cpp:716:73: error: cannot call member function 'int trading_window::get_var_trade(const item&, int)' without object
                         change_amount = max ? ip.charges : get_var_trade( *ip.loc.get_item(), ip.charges );
                                                                         ^
src/npctrade.cpp:725:75: error: cannot call member function 'int trading_window::get_var_trade(const item&, int)' without object
                             change_amount = max ? ip.count : get_var_trade( *ip.loc.get_item(), ip.count );

### Steps To Reproduce

1. Have local repo set to at least 2a5baacbd837207599b0a9017f54a88ded564c00 
2. Run build.

### Expected behavior

Build completes without (breaking) errors.

### Versions and configuration

- OS: Ubuntu 20.04
- Version: 0.F-3065-g951e1d09e1

### Additional context

I was NOT able to get the build failure using the Linux native build line:
make -j$((`nproc`+0)) NATIVE=linux64 CCACHE=1 RELEASE=1 SDL=1 TILES=1 SOUND=1 LTO= PCH= LOCALIZE=1 LINTJSON= ASTYLE= RUNTESTS= LANGUAGES=en

I ran a bisect finding 2a5baacbd837207599b0a9017f54a88ded564c00 is the first bad commit.
I've included buildinfo: 
[buildinfo.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/7255524/buildinfo.zip)

The fact that it's only happening on cross compile leads me to wonder if somehow I've goofed up my mxe setup... again.  That's going to take some time to find out, in the mean time if it's a repo issue the bisect might be handy.

**Comment by ProfoundDarkness** [comment]
OK, figured it out, it was my build environment as I suspected.  There is something missing from the instructions here on getting setup with MXE.  I'm at the end of my day atm (bad time to do anything that might be important) so I'll make sure I've got things sorted out tomorrow and submit a PR to add in the missing detail in the instructions.

Closing this issue as I've fixed it on my end (and the problem was on my end).
